lib/vernier.rb

```diff
@@ -31,9 +31,44 @@ def self.profile(mode: :wall, out: nil, gc: true, **collector_options)
     result
   end
 
+  @collector = nil
+  @collector_out = nil
```

Comment:
(non-blocking) I bet we can move `@collector_out` deeper into the `Collector` so we don't have to maintain this state here. But I'm happy to tackle this in a separate PR.

Reply:
That sounds good to me, I'm happy to address it in a follow up.

Reply:
https://github.com/jhawthorn/vernier/pull/61